core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClient.java

```diff
@@ -270,11 +274,57 @@ protected static ClientRequestContext newDerivedContext(ClientRequestContext ctx
                                                             boolean initialAttempt) {
         final RequestId id = ctx.options().requestIdGenerator().get();
         final EndpointGroup endpointGroup = ctx.endpointGroup();
+        final ClientRequestContext derived;
         if (endpointGroup != null && !initialAttempt) {
-            return ctx.newDerivedContext(id, req, rpcReq, endpointGroup.select(ctx));
+            derived = ctx.newDerivedContext(id, req, rpcReq, endpointGroup.select(ctx));
         } else {
-            return ctx.newDerivedContext(id, req, rpcReq);
+            derived = ctx.newDerivedContext(id, req, rpcReq);
         }
+
+        final RequestLogAccess parentLog = ctx.log();
+        final RequestLog partial = parentLog.partial();
+        final RequestLogBuilder logBuilder = derived.logBuilder();
+        // serializationFormat is always not null, so this is fine.
+        logBuilder.serializationFormat(partial.serializationFormat());
```

Comment:
I wonder that could we move the code of log propopagation to `DefaultRequestLog.addChild(RequestLogAccess)`?

Reply:
As I said before, `addChild` can be called not just by `RetryingClient`, but other places because it's a public API.
We do not know which properties we have to copy from the parent to the child.

Reply:
I understood, thanks for the explanation! 😀